### PR TITLE
Explicitly sort builds according to headings in PR view

### LINKED ISSUE #1480: Builds do not always correspond to headings in PR view
See for instance: https://k8s-gubernator.appspot.com/pr/39204

Here, `bazel` has extra empty `<td>` elements. In addition, the remaining builds are not in the correct order corresponding to the builds, which can be very confusing. For instance, the 2nd and 3rd builds in this row should actually be the first two of the row.

Looking at https://github.com/kubernetes/test-infra/blob/master/gubernator/pull_request.py#L65, it appears that the fact that it usually works OK (e.g, for other jobs here) is a side-effect of the way the input is prepared and the builds for a given job already "pre-sorted".

There is no explicit sort of the builds for a given job in `builds_to_table`. I believe `builds` should be sorted according to the headings before 

```.py
for build, started, finished in builds:
    ...
```

I can prepare a PR if needed.

### PR DESCRIPTION
**Pull request by peay**
Fixes #1480.

This PR sorts the builds explicitly according to:
- the index of the heading they correspond to
- within each heading, by decreasing build number (this is consistent with unit test `test_builds_to_table`)

Added a new test. I only ran unit tests as I do not have a GAE account to test everything with.

edit: I have moved `pad_numbers` to `pull_requests` so we can explicitly sort using build number here.

### CODE REVIEW

**Comment by peay** [comment]
cc @rmmh

File: gubernator/view_base.py [hunk 1]
```diff
@@ -144,8 +143,3 @@ def gcs_ls(path):
     if path[-1] != '/':
         path += '/'
     return list(gcs.listbucket(path, delimiter='/'))
-
-
-def pad_numbers(s):
```
Comment (rmmh) [inline gubernator/view_base.py]: don't move this

File: gubernator/pull_request_test.py [hunk 2]
```diff
@@ -38,6 +44,22 @@ def test_builds_to_table(self):
         self.assertEqual(rows, [('J1', [(4, 'A'), (3, 'B')]),
                                 ('J2', [None, None, (5, 'C'), (4, 'D')])])
 
+        #assert False
```
Comment (rmmh) [inline gubernator/pull_request_test.py]: remove

**Comment by rmmh** [comment]
Is this only a problem with the non-sequential bazel builds? We're going to make all build numbers sequential soon.

File: gubernator/pull_request_test.py (hunk 2 above)
Comment (peay) [inline gubernator/pull_request_test.py]: My bad -- done.

**Comment by peay** [comment]
I've only observed builds in the wrong order with bazel, yes. For the other jobs, it seems sorting by build number before achieves the same effect as sorting by version, so the fix may not be needed for them.

**Comment by rmmh** [comment]
We're going to make bazel build numbers sequential, so this shouldn't be necessary after that.

Thanks for the PR!

**Comment by peay** [comment]
OK, I'll close this then.

**Pull request closed without merging**